### PR TITLE
Flipper auto refresh

### PR DESCRIPTION
**Pull request by asg5704**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *(Summarize the changes that have been made to the platform)*
Implements auto-refresh for Sign in Service in relation to the Flipper UI
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution)*
Call the `/refresh` route on page mount. For subsequent calls, every 4 minutes check for the `vagov_info_token` existence and make a `/refresh` call
- *(Which team do you work for, does your team own the maintainence of this component?)*
Identity. No.
- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*

## Related issue(s)
- *Link to ticket created in va.gov-team repo*
Closes department-of-veterans-affairs/va.gov-team#54147

## Testing done
Manual
- *Describe what the old behavior was prior to the change*
Sign in Service session on Flipper did not work as expected

## What areas of the site does it impact?
Flipper UI

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback
1. Do I need a nonce on the <script> tag?
2. Verify the functionality works as expected
  2a. Authenticate on VA.gov
  2b. Navigating to the Flipper route `/flipper/features`
     - Does a `/refresh` call get made?
     - Waiting for 4 minutes does another `/refresh` call get made?
   2c. Navigating between different Flipper routes does `/refresh` get called?
   2d. Confirm SSOe is not impacted with the Flipper

### CODE REVIEW

File: lib/flipper/views/layout.erb [hunk 1]
```diff
@@ -60,5 +60,13 @@
         <%== yield %>
       </div>
     </div>
+    <script>
+    const baseUrl = '<%= Settings.hostname %>' === '127.0.0.1:3000' ? 'localhost:3000': '<%= Settings.hostname %>';
```
Comment (rileyanderson) [inline lib/flipper/views/layout.erb:64]: Does there need to be a space before the ternary colon? (idk what the JS style is 😅)
```suggestion
    const baseUrl = '<%= Settings.hostname %>' === '127.0.0.1:3000' ? 'localhost:3000' : '<%= Settings.hostname %>';
```

Comment (asg5704) [reply lib/flipper/views/layout.erb:64]: I don't think it matters honestly. I can rewrite it this way too: `const baseUrl = '<%= Settings.hostname %>'.replace('127.0.0.1:3000', 'localhost:3000');`, it just depends on what the BE would prefer 🤷

Comment (rileyanderson) [reply lib/flipper/views/layout.erb:64]: Cool, if you don't think it matters then I don't either

**Review by rileyanderson — APPROVED** [review]
Looks good! Tested and works locally

**Review by bramleyjl — APPROVED** [review]

**Comment by asg5704** [comment]
After trying this out and trying to get it to break, I'm still seeing the Rails error of `Access token is not valid`.  I think more logic will have to be added to Rails but I'm still unsure where.

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**